test/unit/services/proxy_deployment_service_test.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ProxyDeploymentServiceTest < ActiveSupport::TestCase
+  def setup
+    @proxy = FactoryBot.create(:simple_proxy, api_backend: nil)
+  end
+
+  test 'deploy when apicast_configuration_driven' do
+    @proxy.expects(:service_mesh_integration?).returns(false)
+    @proxy.expects(:apicast_configuration_driven).returns(true)
+
+    service = ProxyDeploymentService.new(@proxy)
+    service.expects(:deploy_v2).returns(true)
+
+    assert service.call
+  end
+
+  test 'deploy with service mesh integration' do
+    @proxy.expects(:service_mesh_integration?).returns(true)
+    @proxy.expects(:apicast_configuration_driven).never
+
+    service = ProxyDeploymentService.new(@proxy)
+    service.expects(:deploy_v2).returns(true)
+    service.expects(:deploy_production_v2).returns(true)
+
+    assert service.call
+  end
+
+  test 'deploy production if apicast configuration driven' do
+    @proxy.expects(:apicast_configuration_driven).returns(true)
+
+    service = ProxyDeploymentService.new(@proxy, environment: :production)
+    service.expects(:deploy_v2).never
+    service.expects(:deploy_production_v2).returns(true)
+
+    assert service.call
+  end
+
+  test 'deploy production not apicast configuration driven but api test success' do
+    @proxy.expects(:apicast_configuration_driven).returns(false)
+    @proxy.expects(:api_test_success).returns(true)
+
+    service = ProxyDeploymentService.new(@proxy, environment: :production)
+    service.expects(:deploy_v2).never
```

Comment:
Expecting never is not really testing. Better replace by some expectation: state of an object or creation of proxy config with some attributes testing